### PR TITLE
remove obsolete odb tcl & python

### PR DESCRIPTION
**Pull request by osamahammad21**
closes #803

### CODE REVIEW

File: src/odb/include/odb/db.h [hunk 1]
```diff
@@ -425,7 +420,7 @@ class dbDatabase : public dbObject
   ///
   /// Create an instance of a database
   ///
-  static dbDatabase* create();
+  static dbDatabase* create(utl::Logger* logger = nullptr);
```
Comment (maliberty) [inline src/odb/include/odb/db.h:423]: Same question from the previous PR:  why wouldn't we require a logger.  I don't see why the db should ever create a new one.

**Review by vvbandeira — COMMENTED** [review]
Some notes about documentation

**Comment by maliberty** [comment]
@osamahammad21 is this still relevant?  If so please resolve the conflicts, otherwise close.

File: src/odb/src/swig/common/swig_common.cpp [hunk 2]
```diff
@@ -144,7 +144,7 @@ int write_macro_lef(odb::dbLib* lib, const char* path)
 odb::dbDatabase* read_db(odb::dbDatabase* db, const char* db_path)
 {
   if (db == NULL) {
-    db = odb::dbDatabase::create();
+    db = odb::dbDatabase::create(new utl::Logger());
```
Comment (maliberty) [inline src/odb/src/swig/common/swig_common.cpp:147]: We should use the logger from the app

File: src/odb/test/unitTestsPython/TestITerm.py [hunk 3]
```diff
@@ -51,33 +51,32 @@ def test_connection_from_inst(self):
...
-        mterm_a = self.and2.findMTerm('a')
-        mpin_a = odb.dbMPin_create(mterm_a)
-        geo_box_a_1 = odb.dbBox_create(mpin_a, self.lib.getTech().getLayers()[0], 0, 0, 50, 50)
-        geo_box_a_2 = odb.dbBox_create(mpin_a, self.lib.getTech().getLayers()[0], 0, 0, 100, 100)
-        self.inst.setOrient('R90')
-        self.assertTrue(self.iterm_a.getAvgXY(x, y))
-        self.assertEqual(odb.get_int(x), int( ((0+50)+(0+100))/4 )*-1 )
-        self.assertEqual(odb.get_int(y), int( ((0+50)+(0+100))/4 ) )
+    # def test_avgxy_R0(self):
+    #     x = odb.new_int(0)
```
Comment (maliberty) [inline src/odb/test/unitTestsPython/TestITerm.py:55]: Why is this test disabled?

File: src/odb/src/swig/common/dbtypes_common.i [hunk 4]
```diff
@@ -37,3 +37,9 @@
 
 %apply std::vector<int> &OUTPUT { std::vector<int> & x_grid,
                                   std::vector<int> & y_grid };
+
+%typemap(in, numinputs=0) utl::Logger *INPUT  {
+   $1 = new utl::Logger();
+}
```
[the remark this answers is not in the document]
Comment (osamahammad21) [reply src/odb/src/swig/common/dbtypes_common.i:43]: don't remember why I didn't use Openroad's logger, but will do so.

File: src/odb/test/unitTestsPython/TestITerm.py (hunk 3 above)
Comment (osamahammad21) [reply src/odb/test/unitTestsPython/TestITerm.py:55]: new_int was part of swig for functions that needs passing int by reference. It was removed. we should wrap the function but wasn't of my interest in this PR.

Comment (osamahammad21) [reply src/odb/test/unitTestsPython/TestITerm.py:55]: I think they are wrapped now

**Comment by maliberty** [comment]
Is this still relevant?

**Pull request closed without merging**